### ISSUE TITLE
Remove `svelte/html-self-closing` in the future

### ISSUE THREAD

**Issue by baseballyama** [sveltejs/eslint-plugin-svelte#960]
https://sveltejs.github.io/eslint-plugin-svelte/rules/html-closing-bracket-spacing/

Currently, Svelte 5 determines whether a tag is self-closing based on whether it is a void element and issues warnings accordingly. This enforces correct HTML syntax. Therefore, I believe there is no longer a need to provide a rule for this in ESLint.
https://github.com/sveltejs/svelte/pull/11114

![image](https://github.com/user-attachments/assets/572a86db-4960-4c67-a73d-f02894745f52)

However, this rule is still necessary for certain cases, such as `<Component>`.

**Comment by baseballyama** [comment]
Upon further investigation, it seems that the proper course of action is to improve the implementation of the rule.

**Comment by ota-meshi** [comment]
Yeah, I think we probably need to change the default behavior of the rule.
I think the current options are based on the old prettier-svelte plugin, but should now be changed to the new defaults.
